xbmc/pvr/dialogs/GUIDialogPVRGuideSearch.cpp

```diff
@@ -281,24 +329,49 @@ void CGUIDialogPVRGuideSearch::Update()
   SET_CONTROL_SELECTED(GetID(), CONTROL_BTN_IGNORE_REC, m_searchFilter->ShouldIgnorePresentRecordings());
   SET_CONTROL_SELECTED(GetID(), CONTROL_BTN_IGNORE_TMR, m_searchFilter->ShouldIgnorePresentTimers());
   SET_CONTROL_SELECTED(GetID(), CONTROL_SPIN_NO_REPEATS, m_searchFilter->ShouldRemoveDuplicates());
+  SET_CONTROL_SELECTED(GetID(), CONTROL_BTN_IGNORE_FINISHED,
+                       m_searchFilter->ShouldIgnoreFinishedBroadcasts());
+  SET_CONTROL_SELECTED(GetID(), CONTROL_BTN_IGNORE_FUTURE,
+                       m_searchFilter->ShouldIgnoreFutureBroadcasts());
 
   /* Set time fields */
-  SET_CONTROL_LABEL2(CONTROL_EDIT_START_TIME, m_searchFilter->GetStartDateTime().GetAsLocalizedTime("", false));
+  m_startDateTime = m_searchFilter->GetStartDateTime();
+  if (!m_startDateTime.IsValid())
+  {
+    m_startDateTime = CServiceBroker::GetPVRManager().EpgContainer().GetFirstEPGDate();
+    if (!m_startDateTime.IsValid())
+      m_startDateTime = CDateTime::GetUTCDateTime();
+  }
+
+  m_endDateTime = m_searchFilter->GetEndDateTime();
+  if (!m_endDateTime.IsValid())
+  {
+    m_endDateTime = CServiceBroker::GetPVRManager().EpgContainer().GetLastEPGDate();
+    if (!m_endDateTime.IsValid())
+      m_endDateTime = m_startDateTime + CDateTimeSpan(10, 0, 0, 0);
```

Comment:
How much time you are adding is non obvious from the code. It’s akin to passing many bool values!

Reply:
You mean a should add a comment? I thought everybody is familiar with CDateTimeSpin and knows the semantics of the four parameters. ;-) However, I will add a comment.

Reply:
I rarely interact with it myself. I generally leave the date stuff to others 😉

Reply:
done.